### PR TITLE
Fix #1851 : Fix error 404 when a user leaves a hidden group

### LINKED ISSUE #1851: Error 404 when a user leaves a hidden group
**Describe the bug**
When a user leaves a hidden group, an HTTP 404 error is displayed.

**To Reproduce**
1. Go into a hidden group (where you are a member)
2. Leave that group by clicking on the button "Leave Group"

=> A 404 error is displayed.

**Expected behavior**
When user leaves a hidden group, go back to https://demos.buddyboss.com/online-communities/groups/

**Screenshots**
none

**Support ticket links**
none

### PR DESCRIPTION
**Pull request by sadrul**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

When a regular user leaves a hidden group, a 404 page displays as the page reload. I updated the code to redirect to the groups directory page as suggested in this issue creator

Fixes #1851

### How to test the changes in this Pull Request:

1. Go into a hidden group (where you are a member)
2. Leave that group by clicking on the button "Leave Group"

### Proof Screenshots or Video

https://www.loom.com/share/5ef16adab1e04ab09293522753750fa8

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-templates/bp-nouveau/includes/groups/ajax.php [hunk 1]
```diff
@@ -283,6 +283,10 @@ function bp_nouveau_ajax_joinleave_group() {
 					'type'     => 'success',
 				);
 
+				if ( $group->status == 'hidden' ) {
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/groups/ajax.php:286]: @sadrul Use the strict comparision and other looks good to me.
```suggestion
				if (  'hidden' === $group->status ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]

**Comment by sadrul** [comment]
@chetansatasiya I updated code to use the strict comparison

**Review by chetansatasiya — APPROVED** [review]

**Comment by chetansatasiya** [comment]
@sadrul Merge in dev and add in testing.

**Comment by sadrul** [comment]
@sumitbuddyboss Merged in dev. you can test

**Comment by mohit-buddyboss** [comment]
@sadrul 

FIXED> Tested on dev branch. Now users are redirecting to Group listing page on Leaving Hidden Group.
Video: https://www.loom.com/share/90152bb0767a4bcd84995877cfc8df70

Related issue is resolve now and PR is ready to merge.

@chetansatasiya

**Pull request merged**